### PR TITLE
Add the actual feedback form

### PR DESCRIPTION
**Pull request by vikoivun**
* views with generic class based views
  * somewhat nasty hack to remove the 'not selected' option from the ratings
* vendor a minimal WIP snapshot of hel-bootstrap-4 themes and setup compress
* bring in and templatize the feedback form
* create initial translations for the form
* add a minimal "thank you" page

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -103,3 +103,9 @@ ENV/
 
 # mypy
 .mypy_cache/
+
+# Django static
+/static
+
+# npm
+node_modules/
```
Comment (juyrjola) [inline .gitignore]: Should it have the prepended slash like `/static` above?

File: feedback/templates/feedback/feedback_form.html [hunk 2]
```diff
@@ -0,0 +1,42 @@
+{% extends "minimal.html" %}
+{% load i18n %}
+{% block title %}{{ block.super }} &mdash; {% trans "Give Us Feedback" %}{% endblock title %}
+{% block main %}
+            <div class="feedback-body">
```
Comment (juyrjola) [inline feedback/templates/feedback/feedback_form.html]: Why so much indentation hee?

File: feedback/templates/minimal.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        <meta charset="utf-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+        <title>{% block title %}{% trans "Helsinki feedback" %}{% endblock %}</title>
+{% compress css %}
+        <link rel="stylesheet" type="text/x-scss" href="{% static "feedback-styles.scss" %}">
+{% endcompress %}
+    </head>
+
+    <body>
+
```
Comment (juyrjola) [inline feedback/templates/minimal.html]: Many empty lines here

File: feedback/templates/minimal.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+        <main>
+            {% block main %}{% endblock %}
+        </main>
+
+
+    <!-- jQuery first, then Popper.js, then Bootstrap JS -->
+    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
```
Comment (juyrjola) [inline feedback/templates/minimal.html]: There's a mix of 2-space and 4-space indentation in this file.

File: feedback/templates/feedback/feedback_thankyou.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{% extends "minimal.html" %}
```
Comment (juyrjola) [inline feedback/templates/feedback/feedback_thankyou.html]: The usual convention is to extend `"<app name>/template.html"`, so here `"feedback/minimal.html"`.

File: feedback/templates/feedback/feedback_form.html [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+                <div class="form-group">
+                    <label for="inputFeedbackName" class="control-label">{% trans "Name" %}</label>
+                    <input type="text" class="form-control" name="name" id="inputFeedbackName" placeholder="{% trans "Name" %}">
+                </div>
+                <div class="form-group">
+                    <label for="inputFeedbackEmail" class="control-label">{% trans "Email" %}</label>
+                    <input type="email" class="form-control" name="{{ form.email.html_name }}" id="inputFeedbackEmail" placeholder="{% trans "Email" %}" data-error="{% trans "Please fill in an email address" %}">
+                    <div class="help-block with-errors"></div>
+                </div>
+                <input type="hidden" name="{{ form.url.html_name }}" value="https://linkedevents.hel.ninja/">
```
Comment (juyrjola) [inline feedback/templates/feedback/feedback_form.html]: Linkedevents?

File: feedback/templates/feedback/feedback_form.html [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+                </div>
+                <div class="form-group">
+                    <label for="{{ form.body.id_for_label }}" class="control-label">{% trans "How would you have us improve?" %}</label>
+                    <textarea class="form-control" rows="3" name="{{ form.body.html_name }}" id="{{ form.body.id_for_label }}" required data-error="{% trans "Please fill this field" %}">{% if form.body.value %}{{ form.body.value }}{% endif %}</textarea>
+                    <div class="help-block with-errors"></div>
+                </div>
+                <hr >
+                <p>{% trans "Should you wish for us to contact you, please leave your contact information" %}</p>
+                <div class="form-group">
+                    <label for="inputFeedbackName" class="control-label">{% trans "Name" %}</label>
```
Comment (juyrjola) [inline feedback/templates/feedback/feedback_form.html]: I'd avoid fancyCaps names in HTML

File: feedback/templates/feedback/feedback_thankyou.html [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+{% load i18n %}
+{% block title %}{{ block.super }} &mdash; {% trans "Thank you!" %}{% endblock %}
+{% block main %}
+            <div class="feedback-body text-center">
```
Comment (juyrjola) [inline feedback/templates/feedback/feedback_thankyou.html]: Pretty wide indentation here, too

File: feedback/templates/minimal.html [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.12.9/umd/popper.min.js" integrity="sha384-ApNbgh9B+Y1QKtv3Rn7W3mgPxhU9K/ScQsAP7hUibX39j7fakFPskvXusvfa0b4Q" crossorigin="anonymous"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/js/bootstrap.min.js" integrity="sha384-a5N7Y/aK3qNeh15eJKGWxsqtnX/wWdSZSKp+81YjTmS15nvnvxKHuzaWwXHDli+4" crossorigin="anonymous"></script>
```
Comment (juyrjola) [inline feedback/templates/minimal.html]: Shouldn't we link the 4.0 version here?

File: feedback/views.py [hunk 10]
```diff
@@ -32,3 +34,10 @@ def get(self, request, format=None):
         feedbacks = Feedback.objects.all()
         serializer = FeedbackSerializer(feedbacks, many=True)
         return Response(serializer.data)
+
+
+@method_decorator(xframe_options_exempt, name='dispatch')
+class FeedbackCreate(CreateView):
+    form_class = FeedbackForm
+    model = Feedback
+    success_url = "/thankyou"
```
Comment (juyrjola) [inline feedback/views.py]: Shouldn't this be configured using `reverse()`?

File: feedback/locale/fi/LC_MESSAGES/django.po [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+msgstr "Lähetä"
+
+#: templates/feedback/feedback_thankyou.html:3
+#: templates/feedback/feedback_thankyou.html:6
+msgid "Thank you!"
+msgstr "Kiitos palautteestasi!"
+
+#: templates/minimal.html:8
+msgid "Helsinki feedback"
+msgstr "Helsinki&#45;palaute"
```
Comment (juyrjola) [inline feedback/locale/fi/LC_MESSAGES/django.po]: Why use the HTML character code here?

**Review by juyrjola — CHANGES_REQUESTED** [review]

File: feedback/locale/fi/LC_MESSAGES/django.po (hunk 11 above)
Comment (vikoivun) [reply feedback/locale/fi/LC_MESSAGES/django.po]: Ah. Seems to have been a searching mishap. I though I should use a hyphen here (which is hard to type), but dash &mdash; (—) appears to be the correct one to use.

File: feedback/templates/feedback/feedback_form.html (hunk 6 above)
Comment (vikoivun) [reply feedback/templates/feedback/feedback_form.html]: Ah. Just a placeholder (as the field is mandatory) while waiting for the surrounding widget to finish.

File: feedback/templates/feedback/feedback_form.html (hunk 7 above)
Comment (vikoivun) [reply feedback/templates/feedback/feedback_form.html]: Oh dear. Need to consult with @terotic on this.

File: feedback/views.py (hunk 10 above)
Comment (vikoivun) [reply feedback/views.py]: True. Let's fix that.

File: feedback/templates/feedback/feedback_form.html (hunk 7 above)
Comment (terotik) [reply feedback/templates/feedback/feedback_form.html]: Aren't form field IDs camel case because they map directly into variables on submit?

Comment (terotik) [reply feedback/templates/feedback/feedback_form.html]: Oh, that's the name field that gets submitted. It has been a while I have processed forms. 

I'm just trying to find reasoning why Bootstrap example forms consistently use CamelCase on form element ids...

Comment (vikoivun) [reply feedback/templates/feedback/feedback_form.html]: That would be "name". "id" is used by CSS and JS. Funny, now that i think about it.

File: .gitignore (hunk 1 above)
Comment (vikoivun) [reply .gitignore]: Yes.

File: feedback/locale/fi/LC_MESSAGES/django.po (hunk 11 above)
Comment (vikoivun) [reply feedback/locale/fi/LC_MESSAGES/django.po]: Wait. I was thinking of the wrong place. Hyphen is actually the correct character here. The number is the only way to reliably input it.

Comment (vikoivun) [reply feedback/locale/fi/LC_MESSAGES/django.po]: And given some sensibility training I've decided to repent my ways and use the normal ascii minus sign.

File: feedback/templates/feedback/feedback_form.html (hunk 2 above)
Comment (vikoivun) [reply feedback/templates/feedback/feedback_form.html]: It matches the indentation in the parent thing. Is that a bad thing?

File: feedback/templates/feedback/feedback_thankyou.html (hunk 8 above)
Comment (vikoivun) [reply feedback/templates/feedback/feedback_thankyou.html]: Same thing.  Causes the end product to be pretty.

File: feedback/templates/minimal.html (hunk 3 above)
Comment (vikoivun) [reply feedback/templates/minimal.html]: Let's remove them.

File: feedback/templates/feedback/feedback_form.html (hunk 2 above)
Comment (terotik) [reply feedback/templates/feedback/feedback_form.html]: Interested. I have seen this reasoning before and I have used it myself. But I guess the child should not be aware of the parents' indentation?

File: feedback/templates/minimal.html (hunk 9 above)
Comment (vikoivun) [reply feedback/templates/minimal.html]: Let's see if anything breaks.

File: feedback/templates/feedback/feedback_thankyou.html (hunk 5 above)
Comment (vikoivun) [reply feedback/templates/feedback/feedback_thankyou.html]: Done.

File: feedback/templates/minimal.html (hunk 4 above)
Comment (vikoivun) [reply feedback/templates/minimal.html]: Fixed.

File: feedback/templates/minimal.html (hunk 9 above)
Comment (vikoivun) [reply feedback/templates/minimal.html]: Done too.

File: feedback/views.py (hunk 10 above)
Comment (vikoivun) [reply feedback/views.py]: Done also.

**Comment by terotik** [comment]
Assuming that common consensus is that this is ok now.

**Pull request merged**